From f69da885b88fdb3bf5f2b8e8d29af528881c007e Mon Sep 17 00:00:00 2001
From: Robin Gong <yibin.gong@nxp.com>
Date: Wed, 21 Dec 2016 19:30:06 +0800
Subject: [PATCH 1378/5242] MLK-13636-2 regulator: pf1550-regulator-rmpsg:
 enable pm_qos to prevent low power idle

commit  c7d462b951aee9cc5dd256cb959723bbfaf2d245 from
https://source.codeaurora.org/external/imx/linux-imx.git

CLKE bit of MU_CR may bring VLS state mess in M4 side, so use pm_qos instead
to prevent i.mx7ulp A7 core enter STOP mode during pf1550 send command by rpmsg

Signed-off-by: Robin Gong <yibin.gong@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/regulator/pf1550-regulator-rpmsg.c |    7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/regulator/pf1550-regulator-rpmsg.c b/drivers/regulator/pf1550-regulator-rpmsg.c
index 0bebc7e..a26442e 100644
--- a/drivers/regulator/pf1550-regulator-rpmsg.c
+++ b/drivers/regulator/pf1550-regulator-rpmsg.c
@@ -22,6 +22,7 @@
 #include <linux/module.h>
 #include <linux/of.h>
 #include <linux/platform_device.h>
+#include <linux/pm_qos.h>
 #include <linux/regulator/driver.h>
 #include <linux/regulator/machine.h>
 #include <linux/regulator/of_regulator.h>
@@ -66,6 +67,7 @@ struct pf1550_regulator_info {
 	struct device *dev;
 	struct pf1550_regulator_rpmsg *msg;
 	struct completion cmd_complete;
+	struct pm_qos_request pm_qos_req;
 	struct regulator_desc *regulators;
 };
 
@@ -97,8 +99,13 @@ static int pf1550_send_message(struct pf1550_regulator_rpmsg *msg,
 		return -EINVAL;
 	}
 
+	pm_qos_add_request(&info->pm_qos_req, PM_QOS_CPU_DMA_LATENCY, 0);
+
 	err = rpmsg_send(info->rpdev->ept, (void *)msg,
 			    sizeof(struct pf1550_regulator_rpmsg));
+
+	pm_qos_remove_request(&info->pm_qos_req);
+
 	if (err) {
 		dev_err(&info->rpdev->dev, "rpmsg_send failed: %d\n", err);
 		return err;
-- 
1.7.9.5

